rs/nns/governance/src/storage/neurons.rs

```diff
@@ -343,10 +343,123 @@ where
     /// Returns the next neuron_id equal to or higher than the provided neuron_id
     pub fn range_neurons<R>(&self, range: R) -> impl Iterator<Item = Neuron> + '_
     where
-        R: RangeBounds<NeuronId>,
+        R: RangeBounds<NeuronId> + Clone,
     {
-        self.main.range(range).map(|(neuron_id, neuron)| {
-            self.reconstitute_neuron(neuron_id, neuron, NeuronSections::all())
+        let first = match range.start_bound() {
+            RangeBound::Included(start) => Some(start.clone()), // Start is inclusive, extract value
+            RangeBound::Excluded(start) => Some(start.clone()), // Start is exclusive, extract value
+            RangeBound::Unbounded => Some(NeuronId::MIN),       // No lower bound, return None
+        }
+        .unwrap();
+        // Get the last bound (end of the range)
+        let last = match range.end_bound() {
+            RangeBound::Included(end) => Some(end.clone()), // End is inclusive, extract value
+            RangeBound::Excluded(end) => Some(end.clone()), // End is exclusive, extract value
+            RangeBound::Unbounded => Some(NeuronId::MAX),   // No upper bound, return None
+        }
+        .unwrap();
+
+        let simple_1 = (first, u64::MIN)..(last, u64::MAX);
+        let simple_2 = (first, u64::MIN)..(last, u64::MAX);
+
+        let followees_range = FolloweesKey {
+            follower_id: first,
+            ..FolloweesKey::MIN
+        }..FolloweesKey {
+            follower_id: last,
+            ..FolloweesKey::MAX
+        };
+
+        let mut hot_keys_iter = self.hot_keys_map.range(simple_1).peekable();
```

Comment:
using range on all of these iterators (instead of just the main one) gives us 

instructions: 24.01 M (improved by 38.41%) versus 
 instructions: 38 977 296 (with the old reconstitute_neurons approach)